### PR TITLE
Remove Cloudflare instruction from no data page

### LINKED ISSUE #21752: Remove Cloudflare from No Data Screen
### What happened?

Cloudflare has deprecated their support for "cloudflare apps", see https://cloudflareapps.com/apps/
This means the Matomo Cloudflare app can no longer be used to install the Matomo tracking code.

### What should happen?

The Cloudflare option should be removed from the No Data Screen

### How can this be reproduced?

Not applicable.

### Matomo version

5.0.0

### PHP version

_No response_

### Server operating system

_No response_

### What browsers are you seeing the problem on?

_No response_

### Computer operating system

_No response_

### Relevant log output

_No response_

### Validations

- [X] Read our [Contributing Guidelines](https://github.com/matomo-org/matomo/blob/5.x-dev/CONTRIBUTING.md).
- [X] Follow our [Security Policy](https://github.com/matomo-org/matomo/blob/5.x-dev/SECURITY.md).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating duplicates.
- [X] The provided steps to reproduce is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the Bug.

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

Cloudflare has deprecated their support for "cloudflare apps", see https://cloudflareapps.com/apps/
This means the Matomo Cloudflare app can no longer be used to install the Matomo tracking code.

Therefore this PR removes the suggested installation method from the no data screen.

The detection class is still kept, so Matomo is still be able to detect Cloudflare, as those detection might be used by plugins for other purpose.

fixes #21752 

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/SitesManager/Controller.php [hunk 1]
```diff
@@ -371,7 +371,7 @@ private function getCmsInstruction()
 
         $detectedCms = $this->siteContentDetector->getSiteContentDetectionById(reset($detectedCMSes));
 
-        if (null === $detectedCms) {
+        if (null === $detectedCms || empty($detectedCms::getInstructionUrl())) {
```
Comment (mneudert) [inline plugins/SitesManager/Controller.php:382]: Could that potentially hide available instructions, if multiple CMSs are detected, and the first one happens to be Cloudflare?

For example Cloudflare+Drupal. It should be possible to step over the Cloudflare detection and display the next possible candidate if one is available.

Comment (sgiehl) [reply plugins/SitesManager/Controller.php:382]: good point. will check that.

File: plugins/SitesManager/lang/en.json [hunk 2]
```diff
@@ -82,20 +82,8 @@
         "ShowTrackingTag": "View Tracking code",
         "Sites": "Websites",
         "SiteSearchUse": "You can use Matomo to track and report what visitors are searching in your website's internal search engine.",
-        "SiteWithoutDataCloudflareDescription": "You can use the Matomo App in Cloudflare to track data, follow the instructions from %1$sthis guide%2$s.",
         "SiteWithoutDataCloudflareIntro": "If you use Cloudflare, you can start tracking data in Matomo seamlessly using the Matomo App available in Cloudflare.",
```
Comment (mneudert) [inline plugins/SitesManager/lang/en.json]:
```suggestion
```

This one also seems to not be used anymore.

**Review by mneudert — APPROVED** [review]
One more translation we can probably delete, otherwise LGTM.

**Pull request merged**